### ISSUE TITLE
[guess-parser] has incomplete type definitions

### ISSUE THREAD

**Issue by chrisguttandin** [guess-js/guess#62]
Hi, thanks for publishing a new version of `guess-parser` to npm. It works great, when used in JavaScript code. However it can't be used as part of a TypeScript codebase. The type definition file `dist/guess-parser/index.d.ts` currently looks like this: 

```typescript
export { parseRoutes } from './src/parser';
export { detect } from './src/detector';
export { parseRoutes as parseAngularRoutes } from './src/angular';
export * from './src/react';
export * from './src/preact';
```

It refers to the `./src` directory which is not part of the npm package. I hope it is just a matter of configuring the build process. Unfortunately I don't really know how your build process works.

**Comment by chrisguttandin** [comment]
Hi, I investigated the problem a little bit to see how far I can get. I cloned the repo and could successfully run `npm run bootstrap` and `npm run build` on my local machine. I noticed that the folder at `packages/guess-parser/dist/guess-parser/src` was successfully generated and contained all the necessary type declarations. So I figured that the script for publishing the packages is probably causing the error. Lerna seems to only include those files into the published package which are explicitly referenced in the package.json file. In the case of the guess-parser that's only `dist/guess-parser/index.js` and `dist/guess-parser/index.d.ts`.

Now my conclusion is that adding a files property to the package.json file might solve the problem. But of course that's only a guess since I can't test the publishing script.

What about just adding a files property to explicitly include the full dist folder in the published package?

```json
// ...
"files": [ "dist/" ],
// ...
```

**Comment by addyosmani** [comment]
Was there an update on your investigations here, @mgechev?

**Comment by mgechev** [comment]
Yes, the build process does not produce the type definitions in the correct directory. I'll take care of this over the weekend.

**Comment by mgechev** [comment]
@chrisguttandin the type definitions now are part of the package published under v0.1.4. Thanks for reporting the issue and for the suggestions!

**Comment by chrisguttandin** [comment]
Hi @mgechev, sorry for commenting on that closed issue. Just in case you wonder, why I was so impatiently waiting for guess-parser to be usable as a standalone module. I created a command line tool called [angular-prerender](https://github.com/chrisguttandin/angular-prerender) which aims to simplify the build process for static Angular apps. And it relies on your parser to retrieve all routes of an Angular app. Many thanks again.

**Comment by mgechev** [comment]
@chrisguttandin I found `angular-prerender` as a dependency of the `guess-parser`. I hope it's still useful, after the Angular v6 release.

In the next release of Guess.js we'll remove the `guess-parse` dependency of `guess-webpack`, which should not have any impact on `angular-prerender`.

**Comment by chrisguttandin** [comment]
Hi @mgechev, thanks for letting me know. guess-parser is still working without any problems (at least that I know of) with the latest version of Angular.

I worked on angular-prerender for quite a while now. I gradually added it to all my Angular projects to find out what minimal features it needs. But only today I added the documentation. I guess you didn't really need it to understand what it does. :-)
